pkg/expect/encrypt.go

```diff
@@ -53,51 +50,38 @@ func (exp *AppExpectation) applyDatastoreCipher(datastoreConfig *config.Datastor
 }
 
 func (exp *AppExpectation) prepareCipherData(encBlock *evecommon.EncryptionBlock) (*evecommon.CipherBlock, error) {
-	attestData, err := exp.ctrl.CertsGet(exp.device.GetID())
+	// get device certificate from the controller
+	devCert, err := exp.ctrl.GetECDHCert(exp.device.GetID())
 	if err != nil {
-		log.Errorf("cannot get attestation certificates from cloud for %s will use plaintext", exp.device.GetID())
+		return nil, fmt.Errorf("cannot get device certificate from cloud: %w", err)
```

Comment:
Looks like a logic change, we mute error here before. As I understand we may have no certificate before attestation or for api v1, so please set it back to log and return nil error here. But return error instead of nil few lines below.

Reply:
I agree about the ECDH certificate, but what do you mean by "few lines below? if it's about signing cert, then it should be the same as for the ECDH cert, e.g. api v1 - no error

Reply:
Ah, you are right, signing key is also come from APIv2. So please log and return here. Nothing required for lines below.